FAIL: TestContainerdSuite/TestRunTwice

```
#99  49.43 --- FAIL: TestContainerdSuite (11.72s) | 72s
-- | --
752 | #99  49.43 --- PASS: TestContainerdSuite/TestContainerCleanup (1.62s) | 72s
753 | #99  49.43 --- PASS: TestContainerdSuite/TestImportFail (2.09s) | 72s
754 | #99  49.43 --- PASS: TestContainerdSuite/TestImportSuccess (1.72s) | 72s
755 | #99  49.43 --- PASS: TestContainerdSuite/TestRunLogs (0.31s) | 72s
756 | #99  49.43 --- PASS: TestContainerdSuite/TestRunSuccess (0.31s) | 72s
757 | #99  49.43 --- FAIL: TestContainerdSuite/TestRunTwice (1.12s) | 72s
758 | #99  49.43 containerd_test.go:159: | 72s
759 | #99  49.43 Error Trace: containerd_test.go:159 | 72s
760 | #99  49.43 Error: Received unexpected error: | 72s
761 | #99  49.43 unknown | 72s
762 | #99  49.43 github.com/containerd/containerd/errdefs.init.ializers | 72s
763 | #99  49.43 /go/pkg/mod/github.com/containerd/containerd@v1.2.7/errdefs/errors.go:39 | 72s
764 | #99  49.43 runtime.main | 72s
765 | #99  49.43 /toolchain/go/src/runtime/proc.go:188 | 72s
766 | #99  49.43 runtime.goexit | 72s
767 | #99  49.43 /toolchain/go/src/runtime/asm_amd64.s:1337 | 72s
768 | #99  49.43 failed to listen to abstract unix socket "/containerd-shim/talostest/runtwice/shim.sock": listen unix /containerd-shim/talostest/runtwice/shim.sock: bind: address already in use | 72s
769 | #99  49.43 github.com/containerd/containerd/errdefs.FromGRPC | 72s
770 | #99  49.43 /go/pkg/mod/github.com/containerd/containerd@v1.2.7/errdefs/grpc.go:98 | 72s
771 | #99  49.43 github.com/containerd/containerd.(*container).NewTask | 72s
772 | talos-systems/talos#99 49.43 /go/pkg/mod/github.com/containerd/containerd@v1.2.7/container.go:252

```

--- a/internal/app/machined/pkg/system/runner/containerd/containerd_test.go
+++ b/internal/app/machined/pkg/system/runner/containerd/containerd_test.go
@@ -140,8 +140,10 @@ func (suite *ContainerdSuite) TestRunSuccess() {
 }
 
 func (suite *ContainerdSuite) TestRunTwice() {
+	const ID = "runtwice"
+
 	r := containerdrunner.NewRunner(&userdata.UserData{}, &runner.Args{
-		ID:          "runtwice",
+		ID:          ID,
 		ProcessArgs: []string{"/bin/sh", "-c", "exit 0"},
 	},
 		runner.WithLogPath(suite.tmpDir),
@@ -161,7 +163,12 @@ func (suite *ContainerdSuite) TestRunTwice() {
 		suite.Assert().NoError(r.Stop())
 
 		// TODO: workaround containerd (?) bug: https://github.com/docker/for-linux/issues/643
-		time.Sleep(100 * time.Millisecond)
+		for i := 0; i < 100; i++ {
+			time.Sleep(100 * time.Millisecond)
+			if _, err := os.Stat("/containerd-shim/" + containerdNamespace + "/" + ID + "/shim.sock"); err != nil && os.IsNotExist(err) {
+				break
+			}
+		}
 	}
 }
 

